ssl/statem/statem_dtls.c

```diff
@@ -49,76 +49,57 @@ static int dtls_get_reassembled_message(SSL_CONNECTION *s, int *errtype,
 
 static dtls_sent_msg *dtls1_sent_msg_new(size_t msg_len)
 {
-    dtls_sent_msg *msg = NULL;
-    unsigned char *msg_buf = NULL;
+    dtls_sent_msg *msg = OPENSSL_malloc(sizeof(*msg) + msg_len);
 
-    if ((msg = OPENSSL_zalloc(sizeof(*msg))) == NULL)
+    if (msg == NULL)
         return NULL;
 
-    if (msg_len) {
-        if ((msg_buf = OPENSSL_malloc(msg_len)) == NULL) {
-            OPENSSL_free(msg);
-            return NULL;
-        }
-    }
+    memset(msg, 0, sizeof(*msg));
```

Comment:
why separate out the memset here?  Seems like zalloc would be better

Reply:
To avoid unnecessarily clearing the buffer.